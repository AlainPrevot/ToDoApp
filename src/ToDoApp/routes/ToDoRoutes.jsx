import { Navigate, Route, Routes } from "react-router-dom";
import { ModeStyle, ToDoNav } from "../components";
import { DayPage, MonthPage, WeekPage, YearPage } from "../pages";

import "../../sass/routes/ToDoRoutes.sass";
import { useState } from "react";

export const ToDoRoutes = () => {
  const [theme, setTheme] = useState("light");

  const onMode = () => {
    setTheme(theme === "light" ? "dark" : "light");
  };

  return (
    <div className="body" data-theme={theme}>
      <ModeStyle onMode={onMode} />
      <div className="ToDoRoutes">
        <ToDoNav />

        <Routes>
          <Route path="ToDoApp/day" element={<DayPage />} />
          <Route path="ToDoApp/week" element={<WeekPage />} />
          <Route path="ToDoApp/month" element={<MonthPage />} />
          <Route path="ToDoApp/year" element={<YearPage />} />

          <Route path="/" element={<Navigate to="ToDoApp/day" />} />
        </Routes>
      </div>
    </div>
  );
};
