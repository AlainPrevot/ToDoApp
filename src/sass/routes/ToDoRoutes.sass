body
    background-color: #6D9AB3
    overflow: hidden
    height: 100vh
    display: flex
    align-items: center
    justify-content: center
    color: white

.ToDoRoutes
    background-color: #213140
    width: 620px
    height: 700px
    border-radius: 6px
    display: flex
    flex-direction: column
    align-items: center
    position: relative
