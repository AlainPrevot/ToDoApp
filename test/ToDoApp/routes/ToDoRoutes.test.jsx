/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react";
import { ToDoRoutes } from "../../../src/ToDoApp/routes/ToDoRoutes";
import { MemoryRouter } from "react-router-dom";

describe("Pruebas en <ToDoRoutes/>", () => {
  test("Debe de hacer match con el Snapshot", () => {
    const { container } = render(
      <MemoryRouter>
        <ToDoRoutes />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar DayPage por defecto", () => {
    render(
      <MemoryRouter>
        <ToDoRoutes />;
      </MemoryRouter>
    );

    const dayNav = screen.getByText("Day");

    expect(dayNav.classList.contains("active")).toBe(true);
  });

  test("Debe de mostrar WeekPage", () => {
    render(
      <MemoryRouter initialEntries={["/week"]}>
        <ToDoRoutes />;
      </MemoryRouter>
    );

    const weekNav = screen.getByText("Week");

    expect(weekNav.classList.contains("active")).toBe(true);
  });

  test("Debe de mostrar MonthPage", () => {
    render(
      <MemoryRouter initialEntries={["/month"]}>
        <ToDoRoutes />;
      </MemoryRouter>
    );

    const monthNav = screen.getByText("Month");

    expect(monthNav.classList.contains("active")).toBe(true);
  });

  test("Debe de mostrar YearPage", () => {
    render(
      <MemoryRouter initialEntries={["/year"]}>
        <ToDoRoutes />;
      </MemoryRouter>
    );

    const yearNav = screen.getByText("Year");

    expect(yearNav.classList.contains("active")).toBe(true);
  });
});
