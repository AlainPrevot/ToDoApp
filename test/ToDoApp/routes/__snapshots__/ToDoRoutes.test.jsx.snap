// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ToDoRoutes/> Debe de hacer match con el Snapshot 1`] = `
<div>
  <div
    class="body"
    data-theme="light"
  >
    <input
      class="mode"
      id="mode"
      name="modeStyle"
      type="checkbox"
    />
    <label
      class="modeDark"
      for="mode"
    />
    <label
      class="modeLight"
      for="mode"
    />
    <div
      class="ToDoRoutes"
    >
      <nav>
        <a
          aria-current="page"
          class="active"
          href="/ToDoApp/day"
        >
          Day
        </a>
        <a
          class=""
          href="/ToDoApp/week"
        >
          Week
        </a>
        <a
          class=""
          href="/ToDoApp/month"
        >
          Month
        </a>
        <a
          class=""
          href="/ToDoApp/year"
        >
          Year
        </a>
      </nav>
      <div
        class="ToDoContentListAdd"
      >
        <form>
          <input
            autocomplete="off"
            class=" toDoAdd__input"
            name="description"
            placeholder="Add your next task..."
            type="text"
            value=""
          />
          <button
            class="toDoAdd__button"
            type="submit"
          >
            Add
          </button>
        </form>
        <div
          class="toDoContentListAdd__div"
        >
          <ul
            class="toDoList"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
